database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -47,76 +41,95 @@ public void cleanup() {
 
     public int getCount() {
         return mSnapshots.size();
-
     }
+
     public DataSnapshot getItem(int index) {
         return mSnapshots.get(index);
     }
 
+    protected DataSnapshot internalGetItem(int index) {
```

Comment:
Thanks for catching that!!! I completely messed up removing an item: I wasn't even deleting the snapshot! Check it out now: https://github.com/firebase/FirebaseUI-Android/pull/276/files#diff-c7d9ac7c32439f20359555e3db253268R119